### PR TITLE
ARCHBOM-1036: upgrade to edx-drf-extensions 3.0.1

### PR DESCRIPTION
**Pull request by robrap**
**Before Merge:**
- [x] Switch to actual 3.0.1 release of edx-drf-extensions (rather than commit)

Note: this reverts unecessary test fixes that were added to get 3.0.0 to pass.  Instead, fixes were added to 3.0.1.

ARCHBOM-1036

### CODE REVIEW

**Comment by robrap** [comment]
@Ayub-Khan: I wanted to see if my 3.0.1 fix to 3.0.0 would allow ecommerce to pass tests, even without your changes to the unit tests.  This PR reverts those changes.  I'm not clear on what the errors are on this PR are from.

Note: I think 3.0.1 should work in Stage whether or not the tests pass without your fixes, but I am curious if it does make them pass.  We'll need to see if order fulfillment fails on Stage, unless you can get this all working on a Sandbox.

File: ecommerce/tests/mixins.py [hunk 1]
```diff
@@ -84,12 +84,18 @@ def set_user_id_in_social_auth(self, user, user_id=None):
     def generate_jwt_token_header(self, user, secret=None):
         """Generate a valid JWT token header for authenticated requests."""
         secret = secret or settings.JWT_AUTH['JWT_SECRET_KEY']
+        # WARNING: If this test ever fails due to missing 'exp' or 'iat' or 'is_restricted' claims
```
Comment (dianakhuang) [inline ecommerce/tests/mixins.py]: I'm not sure which test is being referred to, here. Is there a log that should be showing somewhere? An error to look for?

File: requirements/base.in [hunk 2]
```diff
@@ -58,3 +58,5 @@ zeep
 git+https://github.com/django-compressor/django-appconf.git@5169ce2c92d9836e0b3ab3ec645727d9d5225d1a#egg=django-appconf
 
 git+https://github.com/python-social-auth/social-app-django.git@c00d23c2b45c3317bd35b15ad1b959338689cef8#egg=social-auth-app-django
+
+-e git+https://github.com/edx/edx-drf-extensions.git@544619600740a62bed8808a2eb890a49d22a2db0#egg=edx-drf-extensions==3.0.1
```
Comment (dianakhuang) [inline requirements/base.in]: Can now be updated to use the PyPI package, right?

File: ecommerce/tests/mixins.py (hunk 1 above)
Comment (robrap) [reply ecommerce/tests/mixins.py]: I updated the message.  Let me know if it is more clear.

File: requirements/base.in (hunk 2 above)
Comment (robrap) [reply requirements/base.in]: Yes.  It is now on PyPI and has been updated.

**Review by dianakhuang — APPROVED** [review]

**Pull request merged**